### ISSUE TITLE
Send e-mail reminder for people to vote if they haven't

### ISSUE THREAD

**Issue by loorm** [citizenos/citizenos-fe-old#317]
Add an option in voting system dialog window and somewhere in topic settings, for admin to send a reminder to topic members that have not voted to come and vote.

In settings this could be a manual option, with the opportunity to include a custom message. In voting system options dialog, this could be just a tick box and a selection drop-down near the deadline setting to determine, how many days prior to the voting deadline an automatic reminder is sent to all non-voted members.

**Comment by loorm** [comment]
Triage 15: we have no automatic processes right now. Let's make this feature manual. Let's record sending event in activity feed. You can enter a text to be sent (perhaps default being suggested) and press a button. @kevincrepin - can you please put this somewhere in our UI. Also requires design for e-mail.

Requires copy texts for e-mails.

**Comment by tiblu** [comment]
@loorm What do you mean by "voting system dialog window"? Is that the "Voting" tab?

**Comment by loorm** [comment]
> @loorm What do you mean by "voting system dialog window"? Is that the "Voting" tab?

In the red circle.
![voting](https://user-images.githubusercontent.com/22661743/69774794-9e358000-119f-11ea-9f20-30a64007a36c.JPG)

**Comment by kevincrepin** [comment]
@loorm 

Couple comments:

- This is currently only available when creating the voting system, right? At least I couldn't find to change the system anymore after it's created. So that begs the question: Should a voting system be able to be changed after it's been created? Maybe we already discussed this and settled this, don't remember.

- In design we've had a "remind voters" action in the options panel. 

- Should we allow the creator to remind the voters at any point in time? It can create a spam-fest of course. If we keep it completely automated, then yes, it can be part of the voting system creation. If we make it manual, then it needs to be available for the creator and I think made visible in the voting component (where the users cast their vote) as well. So the creator sees from there the option to send out a reminder.

**Comment by loorm** [comment]
We need user consent, to be able to send them anything. Right now, we have not asked for this. So we cannot implement this. This is a part of a larger e-mail notification feature pack.

**Comment by loorm** [comment]
Related to notification system user epic. A decision about this user epic in Big Picture Board will be taken at Triage 45. I propose we unblock the issue, if the decision is to continue with notification system development.

**Comment by anettlinno** [comment]
Triage 46. Discussed with Kati and Ilmar and IF we automate operational messages, we don´t need User´s consent. We could provide option for User to send only one reminder before voting ends. User should be able to choose how many days before voting ends s/he wants the system to send out the reminder. 
@kevincrepin we would love to have a beautiful UX ideas and design from you :) 

PS! We don´t have notification system developed yet and we need to look those together.

**Comment by KatiVellak** [comment]
Can confirm the solution of one automatic reminder email being compliant without explicit consent as this is considered operational email (not targeting email). User can only choose when (how many days before) that one reminder shall be sent.

**Comment by kevincrepin** [comment]
Srry it wasn't exactly clear to me. Is it the person who creates the voting system who sets up whether users will receive a notification about the voting deadline...OR is it the participant who can set this up somewhere themselves?

**Comment by anettlinno** [comment]
> Srry it wasn't exactly clear to me. Is it the person who creates the voting system who sets up whether users will receive a notification about the voting deadline...OR is it the participant who can set this up somewhere themselves?

Sorry for being unclear. 
What we meant was that topic admin user (the one who creates the topic and voting) can choose from the voting system window how many days before the voting ends our system sends out automatic reminder email for participants who have not yet voted. 

Some comments:

- Sending out automatic reminder email should be optional. If the voting time is really short (let´s say 2 days) or only small group is participating then it might not be relevant to send out reminder emails;
- If topic creator wants to change the reminder email time then s/he should be able doing it during the time that voting lasts. I see that topic creators might want to change their mind in the situation where they decide to send out infomail or some other separate email with the reminder to vote.

**Comment by TiinaDUX** [comment]
https://www.figma.com/file/KWTHiHwLZJ3ZFpNJfoVRXX/Platform?node-id=1034%3A25417

Maybe there should be standard choices for vote end reminder?
- 1 day
- 2 days
- 3 days
- 1 week
- 2 weeks
- 1 month

**Comment by anettlinno** [comment]
> https://www.figma.com/file/KWTHiHwLZJ3ZFpNJfoVRXX/Platform?node-id=1034%3A25417
> 
> Maybe there should be standard choices for vote end reminder?
> 
> * 1 day
> * 2 days
> * 3 days
> * 1 week
> * 2 weeks
> * 1 month

That looks beautiful, thank you @TiinaDUX!

Question to @tiblu @ilmartyrk -- would it be possible to show user only the choices about how many days are left until the voting ends? By that I mean that if the voting is open for two weeks then we only provide user an opportunity to send the reminder email out 1 day,  2 days, 3 days and 1 week before the voting ends.

**Comment by anettlinno** [comment]
Triage 55. System should have an opportunity to check if the reminder email was sent or not, so that users couldn´t spam people by sending out several reminder emails. 
What happens if user extends voting deadline? Automatic reminder should also be in settings so that admin user would have an opportunity to change the automatic reminder if they postpone voting. Only one automatic reminder will be sent out by our system. We use the same design in settings as proposed to voting window. 

Sending to development together with notification systems #718. Est. dev. time 2 weeks.
Before launching this issue, we do the user tests @BeccaMelhuish

**Comment by ssin1901** [comment]
@ilmartyrk @anettlinno the notifications system does **not** allow users to opt in to receive reminders to vote (neither in the main environment nor in the test environment). User is notified about all actions related to voting if they select this option from the menu but this is not the same as being able to choose to receive notifications when the deadline is close.

**Comment by ilmartyrk** [comment]
@ssin1901 this is completely different thing actually. Theses reminders are automatically sent to all members of topic. There is no opt in/out.

**Comment by ssin1901** [comment]
@ilmartyrk currently this doesn't seem to be working. I expected to receive a voting reminder 24 hours before the time highlighted in the issue below but didn't.

<img width="1310" alt="Screenshot 2023-01-17 at 16 24 11" src="https://user-images.githubusercontent.com/72747107/212924925-4ae7633e-5de3-4648-a272-3c1c5e513099.png">

**Comment by ssin1901** [comment]
@ilmartyrk these numbers look a bit off to me. 
<img width="1440" alt="Screenshot 2023-01-24 at 12 38 22" src="https://user-images.githubusercontent.com/72747107/214270803-e2103a5e-2ab4-44f0-9440-397e09598dec.png">

**Comment by ssin1901** [comment]
Also still no reminder to vote @ilmartyrk 
<img width="1122" alt="Screenshot 2023-01-26 at 13 49 36" src="https://user-images.githubusercontent.com/72747107/214828410-fdbba568-c7cc-4811-941b-75449e48534b.png">

**Comment by ssin1901** [comment]
@ilmartyrk I think this still has to be fixed and then retested?

**Comment by ilmartyrk** [comment]
@ssin1901 it is working for me, can you possibly send me the topic that you are testing with

**Comment by ssin1901** [comment]
@ilmartyrk I got two email reminders with different deadlines for the same topic. 

1. 
<img width="936" alt="Screenshot 2023-05-08 at 18 24 07" src="https://user-images.githubusercontent.com/72747107/236864452-363114e8-8c75-48c9-9343-23c16e0635fe.png">


2.
<img width="916" alt="Screenshot 2023-05-08 at 18 24 15" src="https://user-images.githubusercontent.com/72747107/236864467-1f34454d-6d29-4626-9d04-2b5f68ebcabd.png">

Might have been that I accessed my email or the topic from two different IP addresses but I think the reminder should not show two different times (and a reminder email should only be sent once).
